### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by RomeoKs**
[DEMO LINK](https://RomeoKs.github.io/react_clock/)

### CODE REVIEW

File: src/Clock.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>): void {
+    const { clockName } = this.props;
+
+    if (clockName !== prevProps.clockName) {
+      console.debug(`Renamed from ${prevProps.clockName} to ${clockName}`);
+    }
+  }
```
Comment (Rijo-J) [inline src/Clock.tsx:34]: Could go for `prevProps` also as you did with current
```suggestion
  componentDidUpdate(prevProps: Readonly<Props>): void {
    const { clockName: prevClockName } = prevProps;
    const { clockName } = this.props;

    if (clockName !== prevClockName) {
      console.debug(`Renamed from ${prevClockName} to ${clockName}`);
    }
  }
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
+      hasClock,
+      clockName,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {hasClock && (
+          <Clock clockName={clockName} />
+        )}
```
Comment (Rijo-J) [inline src/App.tsx:60]: 👍

**Review by Rijo-J — APPROVED** [review]
Achievement unlocked: conquered evil lifeCycle Events 🚀 👍

**Review by naziamoff — APPROVED** [review]
LGTM!

**Pull request closed without merging**